### ISSUE TITLE
MSRV on `sha1` v0.10.2 and v0.10.3 is broken

### ISSUE THREAD

**Issue by striezel** [RustCrypto/hashes#395]
When trying to build a project that depends on `sha1` v0.10.2 with rustc 1.48, one gets error messages from cargo. This appears to be an instance of <https://github.com/rust-lang/cargo/issues/10954>, which means the `sha1` crate simply needs to be re-uploaded with stable cargo or a newer nightly cargo.

The error message is:
```
error: failed to download `sha1 v0.10.2`

Caused by:
  unable to get packages from source

Caused by:
  failed to parse manifest at `/root/.cargo/registry/src/github.com-1ecc6299db9ec823/sha1-0.10.2/Cargo.toml`

Caused by:
  feature `resolver` is required

  this Cargo does not support nightly features, but if you
  switch to nightly channel you can add
  `cargo-features = ["resolver"]` to enable this feature
```

**Comment by striezel** [comment]
@newpavlov: Thanks for the attempted fix in #399. :+1:

Unfortunately, this issue is not fixed in sha1 v0.10.3, the new release still contains a `resolver = "2"` line in the generated Cargo.toml: https://docs.rs/crate/sha1/0.10.3/source/Cargo.toml
Error message is basically still the same as shown above, except with a newer version number.

**Comment by striezel** [comment]
The issues seems to be fixed by #401, i. e. in `sha1` v0.10.4.
